### ISSUE TITLE
Testing phase: iteration 13

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#573]
## To start

```
cd ckeditor5
lerna clean
git pull
mgit update
lerna bootstrap
```

## Automated tests

I'll take care of these.

* Chrome

   ```
   Chrome 61.0.3163 (Mac OS X 10.12.6): Executed 6939 of 6945 (skipped 6) SUCCESS (1 min 10.301 secs / 18.514 secs)
   ```

* Firefox

   Can't run tests – it crashes.

   LOL... Worked after second try.

   ```
   Firefox 57.0.0 (Mac OS X 10.12.0): Executed 6939 of 6945 (3 FAILED) (skipped 6) (2 mins 5.221 secs / 36.905 secs)
   ```

   Similar issues than in Safari – e.g. `Document focus() should focus editable with selection FAILED`.

* Safari 11.0

   ```
   Safari 11.0.0 (Mac OS X 10.12.6): Executed 6458 of 6945 (43 FAILED) (skipped 6) (1 min 34.058 secs / 42.936 secs)
   ```

   Various issues, mainly in the engine and UI. One in core and one in utils. Some in the link. It seems that nothing critical, though – mainly problems with tests.

* Edge 16.x

   Edge is constantly crashing around 50% of the tests:

   ```
   Edge 16.16257.0 (Windows 10 0.0.0): Executed 3124 of 6945 (1 FAILED) DISCONNECTED (1 min 28.887 secs / 33.904 secs)
   ```

   However, it's actually nearly all green until this point.

## Manual tests

* Browse through samples available in https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/index.html and https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/index.html
* Try to edit the content more than to find edge cases in which editor breaks. Remember about copy-paste, file upload (only in "Examples"), Spanish-ISO keyboard and using accents, spellchecker, undo (but not too extensive), hidden features (such as keystrokes and autoformatting).

Browsers:

* Chrome: @pomek @Mgsy
* Firefox: @scofalik @pomek
* Safari 11: @oleq @Mgsy
* Edge 16.x: @szymonkups 

Try to spend maximum 2h on this.

## When I can start?

Even now.

## Where to publish results?

Here. You can include links to issues that you reported (hopefully, you won't need to report any :P)

**Comment by Reinmar** [comment]
As for automated tests:

* In Safari I had some errors that I'll want to check.
* FF is... crashing so I can't run most of the tests. I'll need to check which tests exactly crash it. However, the good news is that I can run engine's tests and they are all green (except one).

**Comment by scofalik** [comment]
https://github.com/ckeditor/ckeditor5-engine/issues/1153

**Comment by scofalik** [comment]
https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/inline-editor.html

It is possible to paste an image inside inline editor. I am not sure whether this is a bug or it should be possible.

**Comment by scofalik** [comment]
https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/classic-editor.html

Images do not have right-align button in toolbars.

**Comment by pomek** [comment]
Regarding the image in the first inline editor.

Scenario:

1. Copy an image from the second editor.
2. Paste the image to the first editor.
3. Type something.
4. Paste the image on more time. You should have `<image></image><paragraph></paragaph><image></image>`.
5. Delete content from element between images.
6. Press <kbd>Backspace</kbd> one more time.

One of the images will be removed. The same scenario works properly in `ClassicEditor` (both images are present).

**Comment by pomek** [comment]
It's impossible to typing after an image.

https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/classic-editor.html

Scenario:

1. Copy the image.
2. Select the whole content and remove it.
3. Paste the image.
4. How to type after the image?

It's reproducible with all editors. I guess we are aware of this issue but I wanted to report it.

**Comment by scofalik** [comment]
@pomek It is possible to focus after and before the image but it is a quirk / bug. Because a paragraph is not created and text goes straight to root. This means that you can't use <kbd>Enter</kbd> and in this state, sooner or later editor will crash.

**Comment by oleq** [comment]
So far the only one which drives me crazy is this old one https://github.com/ckeditor/ckeditor5-link/issues/72.

**Comment by Reinmar** [comment]
@pomek reported that there's no `Code` support in https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/autoformat.html.

**Comment by scofalik** [comment]
[FF] Browser steals selection on CLTR+K
https://github.com/ckeditor/ckeditor5-engine/issues/1154

**Comment by Reinmar** [comment]
Haha 😆 

```js
		it( 'should update editor#element after calling the submit() method', () => {
			return createEditor( textarea )
				.then( editor => {
					expect( textarea.value ).to.equal( '' );

					// Submit method is replaced by our implementation.
					expect( form.submit ).to.not.equal( submitStub );
					form.submit();

					expect( textarea.value ).to.equal( '<p>foo</p>' );
					sinon.assert.calledOnce( submitStub );

					// Check if original function was called in correct context.
					sinon.assert.calledOn( submitStub, form );

					return editor.destroy();
				} );
		} );
```

Causes page reloads in Firefox :D Which is not a surprise.

**Comment by scofalik** [comment]
https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/autoformat.html

Doc says: "Numbered list – Start a line with 1. or 1) followed by a space." but it is enough just to start with `1` too. Or any number.

**Comment by Mgsy** [comment]
[Edge] Applying the inline style to the text scrolls the editor.
https://github.com/ckeditor/ckeditor5/issues/574

**Comment by oleq** [comment]
> Images do not have left-align button in toolbars. Same in balloon example.

@scofalik It's by design. We're encouraging a semantic approach with a default image style and side image style. See https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/image.html#Image-styles.

**Comment by scofalik** [comment]
But there is left-align button in inline example.

**Comment by oleq** [comment]
You're right. But still, I'd rather see the inline example simplified that the other way round.

**Comment by Mgsy** [comment]
Applying the link to the header changes its `font-weight`.

## Steps to reproduce
1. Go to https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/examples/builds/inline-editor.html.
2. Select the second header from the first editable.
3. Apply the link.

## Current result
Header's `font-weight` has changed.

## GIF
![bug_cke5](https://user-images.githubusercontent.com/20988892/31071077-fbee1ad0-a762-11e7-94ad-5354f322101a.gif)

**Comment by Mgsy** [comment]
[Docs] Applying the link to the header changes its `font-weight`.

https://github.com/ckeditor/ckeditor5/issues/575

**Comment by scofalik** [comment]
Block elements got merged after changing multiple blocks to a heading
https://github.com/ckeditor/ckeditor5-heading/issues/86

**Comment by scofalik** [comment]
Okay I am done with my testing!

**Comment by Mgsy** [comment]
[Edge] The list displays incorrectly near the left aligned image.
https://github.com/ckeditor/ckeditor5/issues/576

**Comment by ma2ciek** [comment]
https://github.com/ckeditor/ckeditor5/issues/577

**Comment by Reinmar** [comment]
I updated results of automated tests in the initial post.

There are a couple of conclusions:

* We should try to execute tests in a different way – https://github.com/ckeditor/ckeditor5-dev/issues/290.
* Most of the tests I saw failing were red due to some focus and selection issues. Not a surprise because these are the most critical and most unstably implemented pieces of native APIs. The good thing is that we have those issues encapsulated pretty well and they don't affect too many tests.
